source/Octopus.Server.Client/Repositories/Async/ChannelRepository.cs

```diff
@@ -90,15 +90,23 @@ public async Task<ChannelResource> Get(ProjectResource projectResource, string i
         public async Task<ChannelResource> Create(
             ProjectResource projectResource, 
             ChannelResource channelResource,
-            string gitRef = null)
+            string gitRef = null,
+            string commitMessage = null)
         {
             if (!(projectResource.PersistenceSettings is VersionControlSettingsResource settings))
                 return await repository.Channels.Create(projectResource, channelResource);
             
             gitRef = gitRef ?? settings.DefaultBranch;
             
             var link = projectResource.Link("Channels");
-            return await client.Create(link, channelResource, new { gitRef });
+
+            var commitResouce = new CommitResource<ChannelResource>
+            {
+                CommitMessage = commitMessage,
+                Resource = channelResource
+            };
+
+            return await client.Post<CommitResource, ChannelResource>(link, commitResouce, new { gitRef });
```

Comment:
I notice our other Beta repos use `Put` ([example](https://github.com/OctopusDeploy/OctopusClients/blob/master/source/Octopus.Server.Client/Repositories/DeploymentProcessRepository.cs#L71)) - is this endpoint wired up the same way? Might be worth running a test of the client in linqpad to confirm.

Reply:
@emilol but that one is modify and this one is create (I might be missing something)?

Reply:
ah, my bad! Nope not you, I missed that this was a create 😅👍